cosmos/launchpad/construction_hash.go

```diff
@@ -3,21 +3,32 @@ package launchpad
 import (
 	"context"
 	"crypto/sha256"
-	"encoding/base64"
 	"encoding/hex"
 
+	"github.com/cosmos/cosmos-sdk/simapp"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+
 	"github.com/tendermint/cosmos-rosetta-gateway/rosetta"
 
 	"github.com/coinbase/rosetta-sdk-go/types"
 )
 
 func (l launchpad) ConstructionHash(ctx context.Context, req *types.ConstructionHashRequest) (*types.TransactionIdentifierResponse, *types.Error) {
-	bz, err := base64.StdEncoding.DecodeString(req.SignedTransaction)
+	bz, err := hex.DecodeString(req.SignedTransaction)
 	if err != nil {
 		return nil, rosetta.WrapError(ErrInvalidTransaction, "error decoding tx")
 	}
 
-	hash := sha256.Sum256(bz)
+	var stdTx auth.StdTx
+	cdc := simapp.MakeCodec()
```

Comment:
can we create a local codec here instead of importing simapp?

Reply:
yes. We will do this now.

Reply:
I'm thinking we can use simapp once in types and use it everywhere else in crg. Else we may be creating import dependency on multiple sdk modules for crg